AnimatedComponents/src/components/buttons/icon-transition-button/IconTransitionButton.tsx

```diff
@@ -5,20 +5,46 @@ import Animated, { useAnimatedStyle, useSharedValue, withTiming } from 'react-na
 import styles from './iconTransition-button-styles'
 
 interface IIconsTransitionButton {
-  onPress: () => void | boolean | Promise<void>
+  /** buttonStyle : is an optional prop which defines the styles of button. */
   buttonStyle?: ViewStyle
-  startIcon: React.ReactNode
+  /** failedIcon : is a required prop which dictates the icon displayed if the onPress function returns a falsy value. */
   failedIcon: React.ReactNode
-  successIcon: React.ReactNode
+  /** failedLabel : is a required prop which dictates the label to be displayed when transition fails */
+  failedLabel: string
+  /** isDisabled : is an optional prop which dictates that disables the button if true. */
   isDisabled?: boolean
+  /** isLoading : is an optional prop which dictates that disables the button if true, likely to indicate a loading state. */
   isLoading?: boolean
+  /** onPress : is a required prop which dictates the function is called when the button is pressed. */
+  onPress: () => void | boolean | Promise<void>
+  /** startLabel : is a required prop which dictates the label to be displayed initially. */
+  startLabel: string
+  /** startIcon : is a required prop which dictates the initial icon displayed on the button. */
+  startIcon: React.ReactNode
+  /** successIcon : is a required prop which dictates the icon displayed if the onPress function returns a truth value. */
+  successIcon: React.ReactNode
+  /** successLabel : is a required prop which dictates the label to be displayed when transition succeeds */
+  successLabel: string
 }
 
 const IconTransitionButton = (props: IIconsTransitionButton) => {
-  const { onPress, buttonStyle, startIcon, failedIcon, successIcon, isDisabled, isLoading } = props
+  const {
+    buttonStyle,
+    failedIcon,
+    failedLabel,
+    isDisabled,
+    isLoading,
+    onPress,
+    startIcon,
+    startLabel,
+    successIcon,
+    successLabel,
+  } = props
   const translateX = useSharedValue(0)
   const rotationValue = useSharedValue(0)
+  const labelOpacity = useSharedValue(1)
   const [Icon, setIconName] = useState<React.ReactNode>(startIcon)
```

Comment:
We never store components / function in states, instead of taking startIcon as component we can take the name of the icon and here we can use mui icons.